Algorithm.Framework/Portfolio/PortfolioConstructionModel.cs

```diff
@@ -12,12 +12,24 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
 */
+using System;
 using System.Collections.Generic;
+using System.Linq;
+using Accord.Math;
 using QuantConnect.Algorithm.Framework.Alphas;
 using QuantConnect.Data.UniverseSelection;
 
 namespace QuantConnect.Algorithm.Framework.Portfolio
 {
+    public static class Tools
+    {
+        public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key, TValue defaultValue)
```

Comment:
This is an overly generic method that may be better placed in `QuantConnect.Common\Util\LinqExtensions.cs`